canonical/react-components
# feat: add application layout components

## Done

- Add Vanilla application layout components.

Note: this PR requires the follow to land first:
- https://github.com/canonical/react-components/pull/1088
- https://github.com/canonical/react-components/pull/1089

Original PR and discussion: https://github.com/canonical/react-components/pull/1031.

A number of tests were performed using these components:

https://github.com/canonical/lxd-ui/pull/768
https://github.com/canonical/identity-platform-admin-ui/pull/309
https://github.com/canonical/maas-ui/pull/5434
https://github.com/canonical/juju-dashboard/pull/1752
https://github.com/canonical/rebac-admin/pull/106
https://github.com/canonical/anbox-cloud-dashboard/pull/806

## QA

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

Note: the following PRs need to merge before this can be QAed:
- https://github.com/canonical/react-components/pull/1088
- https://github.com/canonical/react-components/pull/1089
- 
- Go to the demo storybook.
- Go to the application layout components and check the docs look OK.

### Percy steps

- New components expected.

## Fixes

Fixes: #420.
